### PR TITLE
Adding managed identity supports to the api

### PR DESCRIPTION
**Pull request by thomasp98296**
# Choose a PR Template

Switch to "Preview" on this description then select one of the choices below.

<a href="?expand=1&template=data_plane_template.md">Click here</a> to open a PR for a Data Plane API.

<a href="?expand=1&template=control_plane_template.md">Click here</a> to open a PR for a Control Plane (ARM) API.

### CODE REVIEW

File: specification/monitor/resource-manager/Microsoft.Insights/preview/2024-10-01-preview/actionGroups_API.json [hunk 1]
```diff
@@ -0,0 +1,1322 @@
...
+            "schema": {
+              "$ref": "#/definitions/ActionGroupResource"
+            },
+            "description": "The action group to create or use for the update."
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ApiVersionParameter"
```
Comment (ms-bogdan-rotaru) [inline specification/monitor/resource-manager/Microsoft.Insights/preview/2024-10-01-preview/actionGroups_API.json:68]: You're using two different versions of common types.

File: specification/monitor/resource-manager/Microsoft.Insights/preview/2024-10-01-preview/actionGroups_API.json [hunk 2]
```diff
@@ -0,0 +1,1322 @@
...
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
```
Comment (ms-bogdan-rotaru) [inline specification/monitor/resource-manager/Microsoft.Insights/preview/2024-10-01-preview/actionGroups_API.json:374]: Same comment about common types version.

File: specification/monitor/resource-manager/Microsoft.Insights/preview/2024-10-01-preview/actionGroups_API.json [hunk 3]
```diff
@@ -0,0 +1,1322 @@
...
+    }
+  },
+  "definitions": {
+    "AzureResource": {
+      "properties": {
+        "id": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Azure resource Id"
+        },
```
Comment (ms-bogdan-rotaru) [inline specification/monitor/resource-manager/Microsoft.Insights/preview/2024-10-01-preview/actionGroups_API.json:448]: format should be set as "arm-id" if this is the fully qualified resource id

File: specification/monitor/resource-manager/Microsoft.Insights/preview/2024-10-01-preview/actionGroups_API.json [hunk 4]
```diff
@@ -0,0 +1,1322 @@
...
+    },
+    "NotificationRequestBody": {
+      "description": "The request body which contain contact detail metadata",
+      "type": "object",
+      "properties": {
+        "alertType": {
+          "type": "string",
+          "maxLength": 30,
+          "description": "The value of the supported alert type. Supported alert type values are: servicehealth, metricstaticthreshold, metricsdynamicthreshold, logalertv2, smartalert, webtestalert, logalertv1numresult, logalertv1metricmeasurement, resourcehealth, activitylog, actualcostbudget, forecastedbudget"
+        },
```
Comment (ms-bogdan-rotaru) [inline specification/monitor/resource-manager/Microsoft.Insights/preview/2024-10-01-preview/actionGroups_API.json:604]: If there is a fixed list of supported values - have you considered modelling this as enum?

File: specification/monitor/resource-manager/Microsoft.Insights/preview/2024-10-01-preview/actionGroups_API.json [hunk 5]
```diff
@@ -0,0 +1,1322 @@
...
+      "type": "object",
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "The name of the email receiver. Names must be unique across all receivers within an action group."
+        },
+        "emailAddress": {
+          "type": "string",
+          "description": "The email address of this receiver."
+        },
```
Comment (ms-bogdan-rotaru) [inline specification/monitor/resource-manager/Microsoft.Insights/preview/2024-10-01-preview/actionGroups_API.json:833]: Any regex pattern enforcement on this property?

**Comment by ms-bogdan-rotaru** [comment]
Please fill in the PR description by choosing the corresponding PR template and also acknowledging the due diligence checklist.

**Comment by ms-bogdan-rotaru** [comment]
Please fix the errors reported by the required checks.

File: specification/monitor/resource-manager/Microsoft.Insights/preview/2024-10-01-preview/actionGroups_API.json (hunk 2 above)
Comment (thomasp98296) [reply specification/monitor/resource-manager/Microsoft.Insights/preview/2024-10-01-preview/actionGroups_API.json:374]: I changed to v3 now. Thank you!

File: specification/monitor/resource-manager/Microsoft.Insights/preview/2024-10-01-preview/actionGroups_API.json (hunk 3 above)
Comment (thomasp98296) [reply specification/monitor/resource-manager/Microsoft.Insights/preview/2024-10-01-preview/actionGroups_API.json:448]: Change to 'arm-id' as long as this is not a breaking changes.

File: specification/monitor/resource-manager/Microsoft.Insights/preview/2024-10-01-preview/actionGroups_API.json (hunk 4 above)
Comment (thomasp98296) [reply specification/monitor/resource-manager/Microsoft.Insights/preview/2024-10-01-preview/actionGroups_API.json:604]: I don't think so. That is going to be a breaking changes. Furthermore, this is not introduced by my current changes.

**Pull request closed without merging**